src/libraries/analytics/analyticsEventTracking.ts

```diff
@@ -32,18 +22,43 @@ function getAnalyticsInstance(userId?: string) {
 
   try {
     if (userId) {
-      SEGMENT_INSTANCE.userInfo.set({
-        anonymousId: userId,
-        userId,
-        traits: {name: profile?.username, id: userId}
-      });
-
-      MIXPANEL_INSTANCE.identify(userId);
-      MIXPANEL_INSTANCE.getPeople().set('$name', profile?.username);
+      if (SEGMENT_INSTANCE) {
+        SEGMENT_INSTANCE.userInfo.set({
+          anonymousId: userId,
+          userId,
+          traits: {name: profile?.username, id: userId}
+        });
+      }
+
+      if (MIXPANEL_INSTANCE) {
+        MIXPANEL_INSTANCE.identify(userId);
+        MIXPANEL_INSTANCE.getPeople().set('$name', profile?.username);
+      }
     }
   } catch (e) {
     console.log('Error getting signed user id', e);
   }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error message 'Error getting signed user id' is not very descriptive and might be misleading as the error could be related to setting the user identity rather than getting the user id. Consider changing the error message to something more descriptive.

```diff
- console.log('Error getting signed user id', e);
+ console.error('Error setting client identity', e);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->